eng/pipelines/templates/jobs/build-azd-extension.yml

```diff
@@ -34,87 +34,36 @@ jobs:
     timeoutInMinutes: 180
     steps:
       - checkout: self
+
       - template: /eng/pipelines/templates/steps/setup-go.yml
         parameters:
           Condition: and(succeeded(), ne(variables['Skip.LiveTest'], 'true'))
 
-      - template: /eng/pipelines/templates/steps/set-cli-version-cd.yml
-
-      - task: PowerShell@2
-        inputs:
-          pwsh: true
-          targetType: filePath
-          filePath: eng/scripts/Set-CliVersionVariable.ps1
-        displayName: Set CLI_VERSION
-
-      - task: PowerShell@2
-        inputs:
-          pwsh: true
-          targetType: filePath
-          filePath: eng/scripts/Set-CliBuildVariables.ps1
-          arguments: -BuildReason $(Build.Reason)
-        displayName: Set CLI build run variables
+      # TODO: DRY this up, also incorporate CI/"daily" builds
+      - pwsh: |
+          $extVersion = Get-Content ./version.txt
+          Write-Host "##vso[task.setvariable variable=EXT_VERSION;]$extVersion"
+        workingDirectory: ${{ parameters.AzdExtensionDirectory }}
+        displayName: Set EXT_VERSION
 
       - task: PowerShell@2
         inputs:
           pwsh: true
           targetType: filePath
-          filePath: cli/azd/extensions/scripts/ci-build.ps1
+          filePath: ${{ parameters.AzdExtensionDirectory }}/ci-build.ps1
```

Comment:
My original intent was to have a shared `ci-build.ps1` that all extensions could use instead of each extension having a copy of the build script, thoughts?

Ultimately, we may have to have different flavors of this per programming language but wanted to reduce the cost of ownership of extension developers.

Reply:
Chatted offline: If there is shared code it should be in the `scripts/` folder and referenced from the `<extension>/ci-build.ps1` script. This makes it so pipelines can operate entirely in the extension's folder. 

However, things should be simple enough today that `ci-build.ps1` would be trivially simple for all scenarios.